source/TestFrameworkShared/Helper.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Reflection;
+
+namespace nanoFramework.TestFramework
+{
+    public static class Helper
+    {
+        private static string GetJoinedParams(object[] data)
+        {
+            var returnString = string.Empty;
+            foreach (var item in data)
+            {
+                returnString += $"{item} | ";
+            }
+
+            return returnString.Substring(0, returnString.Length - 4);
```

Comment:
why 4?

Reply:
I'm appending $"{item} | " for each parameter passed into method. No matter on count of parameters it ends with " | " every time. So we've got for example:
"Parameter1 | Parameter2 | " and i thought it would be nice to remove it from end :)
So " | " - that's 3 and length starts with 1 (not 0) and in substring we are starting at 0. 
I can put comment in there if needed, just let me know.

Reply:
Can you just add a comment then on this to make it clear?

Reply:
done, hope it's clear

Reply:
it's more than clear :-)